Library/Homebrew/cmd/update-report.rb

```diff
@@ -101,6 +104,32 @@ def install_core_tap_if_necessary
     ENV["HOMEBREW_UPDATE_BEFORE_HOMEBREW_HOMEBREW_CORE"] = revision
     ENV["HOMEBREW_UPDATE_AFTER_HOMEBREW_HOMEBREW_CORE"] = revision
   end
+
+  def migrate_legacy_cache_if_necessary
```

Comment:
Doesn't look like the cp -R deletes as it goes along. So it could trigger multi-gigabyte copying over and over.